swap_meet/vendor.py

```diff
@@ -0,0 +1,57 @@
+from .item import Item
+
+class Vendor:
+    def __init__(self, inventory = None):
+        self.inventory = inventory or []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        return [item for item in self.inventory if item.category == category]
+        # items = list(filter(lambda x: x.category == category, self.inventory))
+        
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        self.remove(my_item)
+        other_vendor.add(my_item)
```

Comment:
This is really interesting that the items are removed right out of the inventory lists without identifying `self.inventory` explicitly.